first worked version v.1.0.0

walerikq opened the pull request.

walerikq committed:
createPerson method

walerikq committed:
Dev branch Initial Commit

walerikq committed:
worked_version_#1

walerikq committed:
Correct Dto functional

walerikq committed:
error correction & a few question

Commit:
Check code and write comments

walerikq committed:
worked version v.1.0.0

walerikq committed:
migrateOnMap

zzaharovs commented on src/main/java/com/walerikq/hospital/Dto/PatientDto.java:
```diff
@@ -0,0 +1,28 @@
+    private String patronymic;
+
+    @NotBlank
+    private Byte age;
```
А если челу 130?)))
Я бы не парился и делал Integer, экономия памяти не настолько большая чтобы за это париться

zzaharovs commented on src/main/java/com/walerikq/hospital/repository/PatientRepo.java:
```diff
@@ -0,0 +1,115 @@
+     * должна быть логика хранения и работы с данными
+     */
+
+    private Map<Integer,Patient> patientList = new HashMap<>();
```
Тогда скорее patientMap

zzaharovs commented on src/main/java/com/walerikq/hospital/repository/PatientRepo.java:
```diff
@@ -0,0 +1,115 @@
+        return patientList.get(id);
+    }
+
+    public Map<Integer, Patient> getPatientsWithStatus(@NotNull PatientsStatus patientsStatus) {
```
Три момента
1. Тут не обязательно возвращать мапу, лучше вернуть лист
2. Тебе нет нужды использовать цикл со счетчиком. Рекомендую подумать воот в эту сторону
        patientMap.values().stream()
                .filter(x -> (тут какое то твое условие))
                .collect(Collectors.toList());

zzaharovs commented on src/main/java/com/walerikq/hospital/repository/PatientRepo.java:
```diff
@@ -0,0 +1,115 @@
+     * @return List
+     */
+    public Map<Integer, Patient> getAllPatients() {
+        return patientList;
```
тут тоже достаточно вернуть только лист через метод .values

zzaharovs commented on src/main/java/com/walerikq/hospital/service/PatientService.java:
```diff
@@ -0,0 +1,83 @@
+        if (patientDto.getPatronymic() != null){
+            patientRepo.setPatronymicPerson(patientDto.getId(),patientDto.getPatronymic());
+        }
+        if (patientDto.getAge() > 0){
```
а если эйдж у пациента 0? новорожденный

zzaharovs replied:
И еще момент - если эйдж будет null у тебя тут в пациента null запишется

zzaharovs commented on src/main/java/com/walerikq/hospital/service/PatientService.java:
```diff
@@ -0,0 +1,83 @@
+
+
+    public void changingPatientData(PatientDto patientDto){
+        Patient patient = patientRepo.getPatientById(patientDto.getId());
```
Непонятно зачем ты его здесь вытащил. У тебя дальше используется только репозиторий, этот объект ты не трогаешь

walerikq replied:
Я тоже это после коммита заметил, это от старой реализации осталось

zzaharovs commented on src/main/java/com/walerikq/hospital/service/PatientService.java:
```diff
@@ -0,0 +1,83 @@
+
+    }
+
+
```
Вообще я бы в этом методе вытащил пациента из БД, и поэтапно по одному полю бы сеттил в него значения из dto если в dto они не null

А в конце вызвал бы один метод репозитория savePatient куда передается пациент целиком

У тебя получается что ты БД дергаешь 6 раз чтобы каждый раз сохранить в нее одного пациента
Логичнее будет если ты ее дернешь два - чтобы вытащить пациента и чтобы сохранить его обратно обновленного

zzaharovs commented on src/main/java/com/walerikq/hospital/repository/PatientRepo.java:
```diff
@@ -0,0 +1,115 @@
+    /**
+     * Изменение имени клиента
+     * @param name
+     */
```
Это все я бы заменил на один метод savePatient()

walerikq committed:
fix:
add method savePatient for saving DB patients;
delete method setName(Surname,Patronymic,Age,Diseases,Status) in PatientMap.java because add method savePatient;
method changingPatientData upgrade for optimize the work with the DB(6 request in DB replace by 2)
the data type of the Age parameter Byte replaced by Integer and age restrictions are set in Patient.java and PatientDto.java

walerikq committed:
fix 2:
the getPatientsWithStatus and getAllPatients method's now returns a list

zzaharovs commented on src/main/java/com/walerikq/hospital/repository/PatientMap.java:
```diff
@@ -0,0 +1,83 @@
+    public Patient getPatientById(int id) {
+        return patientMap.get(id);
+    }
+
```
тут можно просто return без объявления переменной
ну и закомменченное я бы почистил

zzaharovs commented on src/main/java/com/walerikq/hospital/repository/PatientMap.java:
```diff
@@ -0,0 +1,83 @@
+@AllArgsConstructor
+@NoArgsConstructor
+@Repository
+public class PatientMap {
```
Класс удачнее назвать PatientRepository
Тебе же по сути не важно как он хранит данные и откуда берет

zzaharovs commented on src/main/java/com/walerikq/hospital/repository/PatientMap.java:
```diff
@@ -0,0 +1,83 @@
+     *
+     * @return List
+     */
+    public Map<Integer, Patient> getAllPatients() {
```
Тут я бы возвращал лист
Просто вэльюс.стрим.коллект
Кстати по джава конвеншн после стрима вызов каждой функции с новой строчки
.values().stream()
.collect()

walerikq committed:
final fix
